coderd/users.go

```diff
@@ -1596,6 +1596,11 @@ func convertAPIKey(k database.APIKey) codersdk.APIKey {
 		scopes = append(scopes, codersdk.APIKeyScope(s))
 	}
 
+	allowList := db2sdk.List(k.AllowList, db2sdk.APIAllowListTarget)
+	if len(allowList) == 0 {
+		allowList = append(allowList, codersdk.AllowAllTarget())
```

Comment:
I still think we should never have an empty allow list. The db migration handles this iirc, so defaulting to `all` if the list is empty feels like it's hiding some other issue.